core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/DataRowApi.kt

```diff
@@ -74,17 +74,106 @@ public operator fun AnyRow.contains(column: KProperty<*>): Boolean = containsKey
+/**
+ * {@include [DiffDocs]}
+ */
+public fun <T> DataRow<T>.diff(firstRowResult: Int, expression: RowExpression<T, Int>): Int =
+    prev()?.let { p -> expression(this, this) - expression(p, p) } ?: firstRowResult
+
+/**
+ * {@include [DiffDocs]}
+ */
+public fun <T> DataRow<T>.diff(firstRowResult: Long, expression: RowExpression<T, Long>): Long =
+    prev()?.let { p -> expression(this, this) - expression(p, p) } ?: firstRowResult
+
+/**
+ * {@include [DiffDocs]}
+ */
+public fun <T> DataRow<T>.diff(firstRowResult: Float, expression: RowExpression<T, Float>): Float =
+    prev()?.let { p -> expression(this, this) - expression(p, p) } ?: firstRowResult
+
+@OptIn(ExperimentalTypeInference::class)
+@OverloadResolutionByLambdaReturnType
+/**
+ * {@include [DiffOrNullDocs]}
+ */
+public fun <T> DataRow<T>.diffOrNull(expression: RowExpression<T, Double>): Double? =
+    prev()?.let { p -> expression(this, this) - expression(p, p) }
+
+/**
+ * {@include [DiffOrNullDocs]}
+ */
+public fun <T> DataRow<T>.diffOrNull(expression: RowExpression<T, Int>): Int? =
+    prev()?.let { p -> expression(this, this) - expression(p, p) }
+
+/**
+ * {@include [DiffOrNullDocs]}
+ */
+public fun <T> DataRow<T>.diffOrNull(expression: RowExpression<T, Long>): Long? =
+    prev()?.let { p -> expression(this, this) - expression(p, p) }
+
+/**
+ * {@include [DiffOrNullDocs]}
+ */
+public fun <T> DataRow<T>.diffOrNull(expression: RowExpression<T, Float>): Float? =
+    prev()?.let { p -> expression(this, this) - expression(p, p) }
+
 @OptIn(ExperimentalTypeInference::class)
 @OverloadResolutionByLambdaReturnType
+@Deprecated(
+    "Function was deprecated in favor of new name explicitly stating nullable return value and a new non-null overload",
+    ReplaceWith("this.diffOrNull(expression)", "org.jetbrains.kotlinx.dataframe.api.diffOrNull"),
+    DeprecationLevel.WARNING
+)
 public fun <T> DataRow<T>.diff(expression: RowExpression<T, Double>): Double? =
     prev()?.let { p -> expression(this, this) - expression(p, p) }
 
+@Deprecated(
+    "Function was deprecated in favor of new name explicitly stating nullable return value and a new non-null overload",
+    ReplaceWith("this.diffOrNull(expression)", "org.jetbrains.kotlinx.dataframe.api.diffOrNull"),
+    DeprecationLevel.WARNING
+)
 public fun <T> DataRow<T>.diff(expression: RowExpression<T, Int>): Int? =
     prev()?.let { p -> expression(this, this) - expression(p, p) }
 
+@Deprecated(
+    "Function was deprecated in favor of new name explicitly stating nullable return value and a new non-null overload",
+    ReplaceWith("this.diffOrNull(expression)", "org.jetbrains.kotlinx.dataframe.api.diffOrNull"),
+    DeprecationLevel.WARNING
+)
 public fun <T> DataRow<T>.diff(expression: RowExpression<T, Long>): Long? =
     prev()?.let { p -> expression(this, this) - expression(p, p) }
 
+@Deprecated(
+    "Function was deprecated in favor of new name explicitly stating nullable return value and a new non-null overload",
```

Comment:
starting? not ease to read the sentense, please to reformulate. 

Please, if you use the same string many times (more than one), move it to string constants somewhere and name shortly)

Reply:
He wrote "stating", not "starting"

Reply:
I replaced it with constant string and changed the sentence a bit, you can check it out